### PR TITLE
feat(install_panels): add NTP servers panel

### PR DESCRIPTION
**Pull request by FrankYang0529**
related https://github.com/harvester/harvester/issues/935

### CODE REVIEW

**Comment by bk201** [comment]
Please don't use [linking keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) in the PR description/commit/comments. We want some verification before closing the issue.

**Comment by bk201** [comment]
I wonder if we should still allow users to finish the installation if the NTP server can't be reached.
If the cluster is in an air-gapped environment, the NTP server can't be reached.

**Comment by FrankYang0529** [comment]
Probably, we can just valid whether the domain is FQDN. Also, we can allow users to input empty string.

**Comment by bk201** [comment]
The NTP server can be an IP address (e.g. A local server).
How about displaying a warning dialog to the user if the entered server can't be reached, but still allow the user to continue?


> Probably, we can just validate whether the domain is a FQDN. Also, we can allow users to input empty string.

File: pkg/console/util.go [hunk 1]
```diff
@@ -79,6 +80,35 @@ func validatePingServerURL(url string) error {
 	})
 }
 
+func validateNTPServers(ntpServerList []string) error {
+	if len(ntpServerList) == 0 {
+		return fmt.Errorf("NTP Server should not be empty.")
+	}
```
Comment (gitlawr) [inline pkg/console/util.go:86]: Should it be optional? We tend to show optional inputs after the required ones so that users can easily skip them

Comment (FrankYang0529) [reply pkg/console/util.go]: It's optional now and I put it after the password panel. Thank you.

**Comment by FrankYang0529** [comment]
Hi @bk201 @gitlawr , I updated it. Please help me review again. Thank you 😄 .

**Comment by bk201** [comment]
Looks good! But there is a minor issue:
- Enter an NTP server that can't be reached
- It prompts an error and I can press enter to conintue.
- Press ESC to go back
- If I press enter again, the NTP server is not validated again.

**Comment by FrankYang0529** [comment]
I will reset `c.config.OS.NTPServers` when users press ESC. Thank you.

**Comment by bk201** [comment]
> I will reset `c.config.OS.NTPServers` when users press ESC. Thank you.

I meant going back from Proxy page to NTP page. The validation is not performed.

**Comment by gitlawr** [comment]
>If I press enter again, the NTP server is not validated again.

Seems like the problem still exists if I go back from the proxy page.

**Comment by FrankYang0529** [comment]
@bk201 @gitlawr  Users already press Enter to ignore unreachable NTP servers. Do we need to validate it again after users go back from the proxy panel?

**Comment by bk201** [comment]
The user can change his mind and use a different NTP server, it would be good if validation is done again.

**Comment by FrankYang0529** [comment]
> The user can change his mind and use a different NTP server, it would be good if validation is done again.

Yes, I think we already do it now. Like following steps:

* Users input `error.ntp` and press Enter twice to ignore error.
* Users press Esc to go back NTP servers panel.
* Users change `error.ntp` to `another.error.ntp` and press Enter. We check the input again.
* Users press Enter again to go to the proxy panel and press Esc to go back to NTP servers panel.
* Users change `another.error.ntp` to `error.ntp`. We still check the input again here.

So we revalidate when users change the input, but we go through if users have already ignored the error.

**Comment by bk201** [comment]
Screen Recording 2021-09-15 at 12 37 25 PM (1)
Looks like it doesn't validate the server provided in IP form.

**Comment by FrankYang0529** [comment]
> Screen Recording 2021-09-15 at 12 37 25 PM (1)
> Looks like it doesn't validate the server provided in IP form.

Thanks for indicating the error. I send real NTP request now, not only test server connection. Thank you 😄

**Review by gitlawr — APPROVED** [review]
LGTM

**Pull request merged**